src/auth/auth.service.ts

```diff
@@ -0,0 +1,115 @@
+  ERROR_MESSAGES,
+  SUCCESS_MESSAGES,
+} from '@app/common/constants/constants';
+import { JwtService } from '@nestjs/jwt';
+import { CreateUserDto } from './user/dto/create-user.dto';
+import { User } from './user/entity/user.model';
+import { Response } from 'express';
+import { TWO_HOURS_FROM_NOW_DATE } from '@app/common/constants/constants';
+import * as bcrypt from 'bcryptjs';
+import { TokenPayload } from 'src/interfaces/TokenPayload';
+import { ConfigService } from '@nestjs/config';
+@Injectable()
+export class AuthService {
+  constructor(
+    private userService: UserService,
+    private jwtService: JwtService,
+    private configService: ConfigService,
+  ) {}
+
+  async login(loginDto: LoginDto, response: Response) {
+    const dbUser = await this.userService.getUser(loginDto.username);
+    if (!dbUser) {
+      Logger.error('User not found');
+      throw new UnauthorizedException(ERROR_MESSAGES.INVALID_CREDENTIALS);
+    }
+    const isPasswordMatching = await bcrypt.compare(
+      loginDto.password,
+      dbUser.password,
+    );
+    if (!isPasswordMatching) {
+      Logger.error('password not matching');
+      throw new UnauthorizedException(ERROR_MESSAGES.INVALID_CREDENTIALS);
+    }
+    const tokenPayload: TokenPayload = {
+      sub: dbUser.id,
+      username: dbUser.username,
+    };
+    const jwtToken = this.jwtService.sign(tokenPayload);
+
+    const expires = new Date();
+    expires.setSeconds(
+      expires.getSeconds() + this.configService.get('JWT_EXPIRATION'),
+    );
+    response.cookie('Authentication', jwtToken, {
+      httpOnly: true,
+      expires: TWO_HOURS_FROM_NOW_DATE,
+    });
+    return jwtToken;
+  }
+
+  async register(createUserDto: CreateUserDto): Promise<User> {
+    try {
+      const isUserExist =
+        (await this.userService.getUser(createUserDto.username)) !== null;
+      if (isUserExist) {
+        throw new HttpException(
+          ERROR_MESSAGES.USER_ALREADY_EXISTS,
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+      const newUser = await this.userService.createUser(createUserDto);
+      if (!newUser) {
+        Logger.error('Error creating user');
+        throw new HttpException(
+          ERROR_MESSAGES.CREATE_USER,
+          HttpStatus.INTERNAL_SERVER_ERROR,
+        );
+      }
+      return newUser;
+    } catch (error) {
+      Logger.error('Error creating user: ', error);
+      if (
+        error.message.includes('duplicate key value violates unique constraint')
+      ) {
+        throw new HttpException(
+          ERROR_MESSAGES.USER_ALREADY_EXISTS,
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+      if (error.message.includes(ERROR_MESSAGES.USER_ALREADY_EXISTS)) {
+        throw new HttpException(
+          ERROR_MESSAGES.USER_ALREADY_EXISTS,
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+      throw new HttpException(
+        ERROR_MESSAGES.CREATE_USER,
+        HttpStatus.INTERNAL_SERVER_ERROR,
+      );
+    }
+  }
```

Comment:
Also regarding the conditional checks there's no need for so many. You'll throw the proper error in the service and it'll return it here since you've already checked for these errors in the other functions.